src/main/java/seedu/address/model/tier/Tier.java

```diff
@@ -21,14 +21,21 @@ public class Tier {
      */
     public Tier(String tagName) {
         requireNonNull(tagName);
-        checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
-        this.tagName = TierEnum.valueOf(tagName.toUpperCase());;
+        checkArgument(isValidTierName(tagName), MESSAGE_CONSTRAINTS);
+        if (tagName.isEmpty()) {
+            this.tagName = TierEnum.NA;
+        } else {
+            this.tagName = TierEnum.valueOf(tagName.toUpperCase());;
+        }
     }
 
     /**
      * Returns true if a given string is a valid tag name.
      */
-    public static boolean isValidTagName(String test) {
+    public static boolean isValidTierName(String test) {
+        if (test.isEmpty()) {
+            return true;
+        }
```

Comment:
Just to confirm this is for the case where we want to use an empty tier to denote the deleting of a tier from an existing user right?

Reply:
The current logic for `filter` relies on `isValid<Parameter>Name` method to determine if the user input is an acceptable search term. While filtering by tier is yet to be implemented, this change could potentially lead to issues down the road. It will allow users to filter using an empty string, which could lead to Exceptions or match the entire customer base.

I think no action is necessary for now, but I may implement an alternative (albeit slightly redundant) method to handle the validity check for Tier during filter.

Reply:
Further from @itsme-zeix , my intrusive thought is to implement a default/placeholder tier, for a person without an assigned tier. This internal-only classification could streamline our data model and simplify filtering logic, while remaining invisible in the UI. Let's discuss this implication in later.

Reply:
Hi! I was looking at our TierEnum, we actually have:

```
public enum TierEnum {
    GOLD,
    SILVER,
    BRONZE,
    REJECT,
    NA
}
```

The NA enum value likely addresses our need for handling users without a specific tier. This built-in solution should resolve the filtering issue you mentioned. However we might want to confirm if NA is used for users without an assigned tier.

If these points check out, I think we then shouldn't need additional logic for empty tiers. @itsme-zeix